modules/ContextUtils.js

```diff
@@ -13,61 +13,61 @@ function makeContextName(name) {
   return `@@contextSubscriber/${name}`
 }
 
-export function createContextProvider(name) {
+export function ContextProvider(name) {
   const contextName = makeContextName(name)
+  const listenersKey = `${contextName}/listeners`
```

Comment:
If you define something like `[ContextSubscriber('router'), ContextSubscriber('whatever')]`, React would complain that they both define `subscribe` method. Even worse, if that `subscribe` stops being a method, React will blindly merge mixins and thus they would each overwrite the same `this.listeners`.

Finally, the consuming component (such as `<Router>`) may add an instance field called `this.listeners` or a method called `subscribe`. By scoping mixin method names by the corresponding context name, we are playing it safe. We also signal both internally and externally this way that those things (e.g. `this.subscribe`) are not supposed to be called directly even though technically they are available on the `<Router>` instance.